### PR TITLE
Implement new http base connector

### PR DESCRIPTION
**Pull request by cutoffthetop**
- implement new base class for http connectors
- use it in wikidata, backend_api and public_api

### CODE REVIEW

File: mex/common/connector/http.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import requests
+from requests import HTTPError, RequestException, Response
+
+from mex.common.connector import BaseConnector
+from mex.common.settings import BaseSettings
+from mex.common.transform import MExEncoder
+
+
+class HTTPConnector(BaseConnector):
+    """Create a new HTTP connection."""
```
Comment (cutoffthetop) [inline mex/common/connector/http.py:16]:
```suggestion
    """Base class for requests-based HTTP connectors."""
```

File: mex/common/wikidata/connector.py [hunk 2]
```diff
@@ -1,48 +1,20 @@
...
+        self.request("GET", self.url, params={"format": "json"})
 
-    def _get_json_from_api(
-        self, url: str, params: dict[str, Any]
-    ) -> dict[str, dict[str, str]]:
-        response = self.session.get(url=url, params=params, timeout=self.TIMEOUT)
-
-        return cast(dict[str, dict[str, str]], response.json())
-
-    @cache
```
Comment (cutoffthetop) [inline mex/common/wikidata/connector.py:45]: why remove the cache?

File: poetry.lock [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.5.0 and should not be changed by hand.
```
Comment (cutoffthetop) [inline poetry.lock:1]: could you also bump the version of mex-common in the toml please?

File: tests/backend_api/test_connector.py [hunk 4]
```diff
@@ -61,20 +55,22 @@ def test_request_success(
     expected_response: dict[str, Any],
     expected_kwargs: dict[str, Any],
 ) -> None:
-    mocked_send_request = MagicMock(
-        name="_send_request",
-        spec=BackendApiConnector._send_request,
-        return_value=mocked_response,
-    )
-    monkeypatch.setattr(BackendApiConnector, "_send_request", mocked_send_request)
+    mocked_session = MagicMock(spec=requests.Session, name="backend_session")
```
Comment (cutoffthetop) [inline tests/backend_api/test_connector.py:58]: this test is not really backend_api specific anymore (it's only testing logic from http-connector). could you move it to `test_http.py` or remove it if it's redundant

Comment (cutoffthetop) [reply tests/backend_api/test_connector.py:58]: same goes for test_request_error_mocked

**Review by cutoffthetop — COMMENTED** [review]
looks nice, just a couple of smaller comments

File: tests/backend_api/test_connector.py (hunk 4 above)
Comment (mr-kamran-ali) [reply tests/backend_api/test_connector.py]: updated in 100b4c54e8e96cc2cb59169b31594dbc0a08646a

File: poetry.lock (hunk 3 above)
Comment (cutoffthetop) [reply poetry.lock:1]: you marked this resolved, but i don't see this in the diff

Comment (mr-kamran-ali) [reply poetry.lock:1]: I am not sure why its not popping up in the diff, but mex-common version in pyproject.toml is 0.11.0 https://github.com/robert-koch-institut/mex-common/blob/100b4c54e8e96cc2cb59169b31594dbc0a08646a/pyproject.toml#L3C19-L3C19

and lock file also says its created by poetry 1.5.1

**Pull request merged**